pillarbox-demo-shared/src/main/java/ch/srgssr/pillarbox/demo/shared/ui/components/PillarboxSlider.kt

```diff
@@ -0,0 +1,466 @@
+        startInteraction?.let {
+            interactionSource.tryEmit(finalInteractionConstructor(it))
+            startInteraction = null
+        }
+
+        onSliderValueChangeFinished()
+    }
+
+    return this then Modifier.pointerInput(Unit) {
+        detectHorizontalDragGestures(
+            onDragStart = { offset ->
+                startInteraction = DragInteraction.Start()
+                    .also { interactionSource.tryEmit(it) }
+
+                onSliderValueChange(offset.x / size.width)
+            },
+            onDragEnd = {
+                destroyStartInteraction(DragInteraction::Stop)
+            },
+            onDragCancel = {
+                destroyStartInteraction(DragInteraction::Cancel)
+            },
+            onHorizontalDrag = { change, _ ->
+                onSliderValueChange(change.position.x / size.width)
+            },
+        )
+    }
+}
+
+@Composable
+private fun Track(
+    weight: Float,
+    color: Color,
+    modifier: Modifier = Modifier,
+) {
+    if (weight > 0f) {
+        Box(
+            modifier = modifier
+                .fillMaxHeight()
+                .fillMaxWidth(fraction = weight)
+                .background(
+                    color = color,
+                    shape = CircleShape,
+                ),
+        )
+    }
+}
+
+@Composable
+private fun Thumb(
+    color: Color,
+    enabled: Boolean,
+    modifier: Modifier = Modifier,
+    onSeekBack: () -> Unit,
+    onSeekForward: () -> Unit,
+) {
+    Box(
+        modifier = modifier
+            .fillMaxHeight()
+            .width(8.dp)
+            .background(
+                color = color,
+                shape = CircleShape,
+            )
+            .then(
+                if (enabled) {
+                    Modifier
+                        .focusable()
+                        .onDpadEvent(
+                            onLeft = {
+                                onSeekBack()
+                                true
+                            },
+                            onRight = {
+                                onSeekForward()
+                                true
+                            },
+                        )
+                } else {
+                    Modifier
+                }
+            ),
+    )
+}
+
+@Composable
+@PreviewLightDark
+private fun PillarboxSliderPreview(
+    @PreviewParameter(PillarboxSliderPreviewParameters::class) previewParameters: PreviewParameters,
+) {
+    val isDark = isSystemInDarkTheme()
+    var progress by remember {
+        mutableLongStateOf(previewParameters.initialValue)
+    }
+
+    PillarboxSlider(
+        value = progress,
+        range = 0L..100L,
+        compactMode = previewParameters.compactMode,
+        secondaryValue = previewParameters.secondaryValue,
+        enabled = previewParameters.enabled,
+        thumbColorEnabled = if (isDark) md_theme_dark_primary else md_theme_light_primary,
```

Comment:
It's because we are in `pillarbox-demo-shared`, where we don't have access to `MaterialTheme`. And we can't add the Material dependency, because then it wouldn't work on both `pillarbox-demo` and `pillarbox-demo-tv`.
Anyway, this is just a preview for the `PillarboxSlider` composable. It's not an actual code part of the demo applications.